File: frontend/lib/models/question.dart
class Question {
  final int questionID;
  final int postID;
  final String questionText;
  final bool isResolved;
  final DateTime? resolvedAt;
  final int? answerID;

  Question({
    required this.questionID,
    required this.postID,
    required this.questionText,
    required this.isResolved,
    this.resolvedAt,
    this.answerID,
  });

  factory Question.fromJson(Map<String, dynamic> json) => Question(
        questionID: json['questionID'],
        postID: json['postID'],
        questionText: json['questionText'],
        isResolved: json['isResolved'] == 1,
        resolvedAt: json['resolvedAt'] != null ? DateTime.parse(json['resolvedAt']) : null,
        answerID: json['answerID'],
      );
}

File: backend/routes/question.js
const express = require('express');
const router = express.Router();
const db = require('../db');

// Get all questions of a post
router.get('/:postID', async (req, res) => {
  const [rows] = await db.query('SELECT * FROM Question WHERE postID = ?', [req.params.postID]);
  res.json(rows);
});

// Add a question
router.post('/', async (req, res) => {
  const { postID, questionText } = req.body;
  const [result] = await db.query(
    `INSERT INTO Question (postID, questionText, isResolved)
     VALUES (?, ?, 0)`,
    [postID, questionText]
  );
  res.status(201).json({ questionID: result.insertId });
});

module.exports = router;
